### PR TITLE
AUTOM-2883 <FilterDropdown/> component

### PR DESCRIPTION
**Pull request by alexabreu**
image

## Merge Checklist

- [x] Includes link to associated Github Issue or JIRA ticket
https://jira.sendgrid.net/browse/AUTOM-2883
- [x] Has descriptive PR title and body
- [x] All changes are backwards compatible
- [x] QA'd by the SendGrid Style Guide designer
- [x] Code reviewed by a SendGrid frontend lead
- [x] If adding a new component, the component has been added to the exports in `src/index.ts`.

### CODE REVIEW

File: src/checkbox.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+import React, { ReactNode } from 'react';
 import Styles from './styles/checkbox-radio.module.scss';
 import cn from './utilities/classnames';
 export interface CheckboxProps {
   checked: boolean;
   disabled?: boolean;
   id: string;
   info?: string;
-  label?: string;
+  label?: ReactNode;
```
Comment (Ditofry) [inline src/checkbox.tsx]: could this cause problems for current consumers of the `Checkbox` component?

Comment (alexabreu) [reply src/checkbox.tsx]: i don't think so. ReactNode is a superset of string.

Comment (alexabreu) [reply src/checkbox.tsx]: Now it's a render prop, so you can pass in a component rather than just a string.

File: src/types/stats.ts [hunk 2]
```diff
@@ -18,3 +18,28 @@ export interface CardStatType {
...
+  | 'requests'
+  | 'spam-report-drops'
+  | 'spam-reports'
+  | 'unique-clicks'
+  | 'unique-opens'
+  | 'unsubscribe-drops'
+  | 'unsubscribes';
+
+export interface FilterDropdownOptionType {
+  type?: CampaignStatType;
```
Comment (Ditofry) [inline src/types/stats.ts]: I'm happy with this but I would also be happy with this being generic

Comment (Ditofry) [reply src/types/stats.ts]: talked about this and I like it 👍

Comment (alexabreu) [reply src/types/stats.ts]: this is strict so the implementor can be assured the passed in type will result in the correctly colored badge within the filter dropdown cmp.

**Comment by thinkingserious** [comment]
Hello @alexabreu,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**